src/core/transaction.c

```diff
@@ -662,7 +662,7 @@ static int transaction_apply(
                         /* j has been merged into a previously installed job */
                         if (tr->anchor_job == j)
                                 tr->anchor_job = installed_job;
-                        hashmap_remove(m->jobs, UINT32_TO_PTR(j->id));
+                        hashmap_remove_value(m->jobs, UINT32_TO_PTR(j->id), j);
```

Comment:
this sounds ok to do, but how can this happen that two jobs have the same id?

Reply:
Without the last two commits, multiple jobs may have zero ID, or have same ID when overflowed.
Of course, that's paranoia, and just for safety.

Note, we already use `hashmap_remove_value()` in `job_uninstall()`.